File: LeetCodeJava/88.合并两个有序数组.java
import java.util.Arrays;

/*
 * @Author: Xun Zhao
 * @Date: 2019-09-21 17:03:28
 * @LastEditors: Xun Zhao
 * @LastEditTime: 2019-09-22 14:47:12
 * @Description: 
 */
/*
 * @lc app=leetcode.cn id=88 lang=java
 *
 * [88] 合并两个有序数组
 */
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int[] tmp = new int[m + n];
        int i = 0;
        int j = 0;
        for (int k = 0; k < m + n; k ++) {
            if (i >= m && j < n) {
                tmp[k] = nums2[j];
                j ++;
            } else if (i < m && j >= n) {
                tmp[k] = nums1[i];
                i ++;
            } else if (nums1[i] <= nums2[j]) {
                tmp[k] = nums1[i];
                i ++;
            } else {
                tmp[k] = nums2[j];
                j ++;
            }
        }
        for (int k = 0; k < m + n; k ++) {
            nums1[k] = tmp[k];
        }
    }
}

File: LeetCodePython/88.合并两个有序数组.py
#
# @lc app=leetcode.cn id=88 lang=python3
#
# [88] 合并两个有序数组
#

# @lc code=start
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        tmp = []
        i = j = 0
        while i < m or j < n:
            if i >= m:
                tmp.append(nums2[j])
                j += 1
            elif j >= n:
                tmp.append(nums1[i])
                i += 1
            elif nums1[i] <= nums2[j]:
                tmp.append(nums1[i])
                i += 1
            else:
                tmp.append(nums2[j])
                j += 1
        for k in range(m + n):
            nums1[k] = tmp[k]
# @lc code=end

